### PR TITLE
CCloud-ify first producer app

### PR DESCRIPTION
**Pull request by ybyzek**
### Description

There is a lot of smaller tweaks/optimizations happening in this PR, and the broad strokes are:

* CCloud-ification of **producer** KT
* Update ccloud-ification of **consumer** -- this was done earlier in a PR from last week but there were some dedupe/cleanup enhancements
* Tweak some of the `shared` ccloud-ification code, including a screenshot from Confluent Cloud

### Staging Docs

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/ccloud-first-producer-app
http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/ccloud-first-producer-app/creating-first-apache-kafka-producer-application/confluent.html
http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/ccloud-first-producer-app/creating-first-apache-kafka-consumer-application/confluent.html
http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/ccloud-first-producer-app/creating-first-apache-kafka-producer-application/kafka.html
http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/ccloud-first-producer-app/creating-first-apache-kafka-consumer-application/kafka.thml

### CODE REVIEW

**Comment by ybyzek** [comment]
@bbejeck @rspurgeon any suggestions on how to address the error?

https://confluentinc.semaphoreci.com/jobs/dde1da8c-a7ed-4015-b196-dd2e9e4024ac

> Tutorial _includes:tutorials:kafka-consumer-application:confluent:code was not included into the build. Check settings.gradle for details.

This does not fail with @awalther28 original PR (https://confluentinc.semaphoreci.com/workflows/9c7c4b79-ca30-4906-a881-3380c8f87ca2?pipeline_id=2b96adbe-8ce0-45e7-89c3-fc35328bd260) because this PR adds `confluent/code`.

**Comment by ybyzek** [comment]
IIUC the failing test is unrelated to changes in this PR

File: _includes/shared/markup/ccloud/ccloud-setup-self.adoc [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
-Next click on `Learn` and follow the in-product instructions to launch Kafka and Schema Registry.
-Use the promo code `CC100KTS` to receive an additional $100 free usage on Confluent Cloud (https://www.confluent.io/confluent-cloud-promo-disclaimer[details]).
+1. After you log in to Confluent Cloud, click on `Add cloud environment` and name the environment `learn-kafka`. Using a new environment keeps your learning resources separate from your other Confluent Cloud resources.
 
+2. From the `Billing & payment` section, apply the promo code `CC100KTS` to receive an additional $100 free usage on Confluent Cloud (https://www.confluent.io/confluent-cloud-promo-disclaimer[details]).
+
+3. Click on `LEARN` and follow the instructions to launch a Kafka cluster and to enable Schema Registry.
+
++++++
+<img src="/assets/img/ccloud-home.png" alt="Confluent Cloud" width=800>
```
Comment (bbejeck) [inline _includes/shared/markup/ccloud/ccloud-setup-self.adoc:11]: The image renders like this 
Screen Shot 2021-04-13 at 3 51 52 PM

The `img` tag needs to be formatted like this:

``

Looks like this content is reused across other tutorials. So fixing this here and rebasing the other PRs should fix them as well.

**Review by bbejeck — CHANGES_REQUESTED** [review]
Overall looks great @ybyzek!  One minor fix noted in the comments.

Also, can you rebase this PR against master to pick up https://github.com/confluentinc/kafka-tutorials/pull/789? It will set the `build.gradle` files to use maven central vs. jcenter.

File: _includes/shared/markup/ccloud/ccloud-setup-self.adoc (hunk 1 above)
Comment (ybyzek) [reply _includes/shared/markup/ccloud/ccloud-setup-self.adoc]: Thanks for the tip @bbejeck .  Trying this new format in https://github.com/confluentinc/kafka-tutorials/pull/790/commits/0128b0978acbac6c727d41609c58570d56dcba98

**Comment by ybyzek** [comment]
> Also, can you rebase this PR against master to pick up #789? It will set the build.gradle files to use maven central vs. jcenter.

@bbejeck given that this PR is merging to `ccloud-master`, which does not have #789, should this still be rebased?  (And if yes, how?, given that #789 went into `master`, not `ccloud-master`?). 

Alternatively, I can work this week's set of 5 KTs into `ccloud-master`, and we'll merge `ccloud-master` into `master` at the end of the week -- but we'll rebase `ccloud-master` at that time before merge.  WDYT about proceeding this way?

**Comment by bbejeck** [comment]
> Alternatively, I can work this week's set of 5 KTs into ccloud-master, and we'll merge ccloud-master into master at the end of the week -- but we'll rebase ccloud-master at that time before merge. WDYT about proceeding this way?

SGTM

File: _includes/shared/markup/ccloud/ccloud-setup-self.adoc (hunk 1 above)
Comment (ybyzek) [reply _includes/shared/markup/ccloud/ccloud-setup-self.adoc]: @bbejeck the image now renders (thanks to your help).

**Review by rspurgeon — APPROVED** [review]
LGTM

**Review by bbejeck — APPROVED** [review]

**Pull request merged**